src/django/api/views.py

```diff
@@ -64,3 +71,85 @@ def total_factories(request):
 @permission_classes((AllowAny,))
 def search_factories(request):
     return Response([])
+
+
+class FacilityListViewSet(viewsets.ModelViewSet):
+    # TODO: Filter based on auth.
+    queryset = FacilityList.objects.all()
+    serializer_class = FacilityListSerializer
+
+    def _parse_csv_line(self, line):
+        return list(csv.reader([line]))[0]
+
+    def _validate_header(self, header):
+        if header is None or header == '':
+            raise ValidationError('Header cannot be blank.')
+        parsed_header = [i.lower() for i in self._parse_csv_line(header)]
+        if 'country' not in parsed_header \
+           or 'name' not in parsed_header \
+           or 'address' not in parsed_header:
+            raise ValidationError(
+                'Header must contain country, name, and address fields.')
+
+    @transaction.atomic
+    def create(self, request):
+        # TODO: Require authenticated user with related organization
+        if 'file' not in request.data:
```

Comment:
Consider setting `'file'`, `'name'`, `'replaces'` and others of these string keys as constants.

Reply:
Constants are a good idea. Thanks.